dbt/adapters/base/impl.py

```diff
@@ -627,6 +628,11 @@ def get_columns_in_relation(self, relation: BaseRelation) -> List[BaseColumn]:
         """Get a list of the columns in the given Relation."""
         raise NotImplementedError("`get_columns_in_relation` is not implemented for this adapter!")
 
+    @abc.abstractmethod
+    def get_relation_metadata(self, relation: BaseRelation) -> Tuple[Any, Any]:
```

Comment:
I'm not a fan of the return type here, but the return type I want `Tuple[Optional[TableMetadata], StatsDict]` is buried in `dbt-core/artifacts` 🥲